conans/test/functional/generators/components/cmake_find_package_multi_test.py

```diff
@@ -804,3 +804,146 @@ def build(self):
+            """)
+        middle_cpp = textwrap.dedent("""
+            #include "middle.h"
+            #include "expected.h"
+            #include "variant.h"
+
+            void middle() {
+                expected();
+                variant();
+            }
+            """)
+        middle_conanfile = textwrap.dedent("""
+            import os
+            from conans import ConanFile, CMake
+
+            class Conan(ConanFile):
+                name = "middle"
+                version = "1.0"
+                settings = "os", "compiler", "build_type", "arch"
+                generators = "cmake", "cmake_find_package_multi"
+                exports_sources = "src/*"
+                requires = "expected/1.0", "variant/1.0"
+
+                def build(self):
+                    cmake = CMake(self)
+                    cmake.configure(source_folder="src")
+                    cmake.build()
+
+                def package(self):
+                    self.copy("*.h", dst="include", src="src")
+                    self.copy("*.lib", dst="lib", keep_path=False)
+                    self.copy("*.a", dst="lib", keep_path=False)
+
+                def package_info(self):
+                    self.cpp_info.libs = ["middle"]
+            """)
+        client.save({"conanfile.py": middle_conanfile, "src/CMakeLists.txt": middle_cmakelists,
+                     "src/middle.h": middle_h, "src/middle.cpp": middle_cpp}, clean_first=True)
+        client.run("create . middle/1.0@")
+        conanfile = textwrap.dedent("""
+            import os
+            from conans import ConanFile, CMake
+
+            class Conan(ConanFile):
+                name = "consumer"
+                version = "1.0"
+                generators = "cmake_find_package_multi", "cmake"
+                settings = "os", "compiler", "build_type", "arch"
+                exports_sources = "src/*"
+                requires = "middle/1.0"
+
+                def build(self):
+                    cmake = CMake(self)
+                    cmake.configure(source_folder="src")
+                    cmake.build()
+                    bin_path = os.path.join("bin", "main")
+                    self.run(bin_path, run_environment=True)
+            """)
+        cmakelists = textwrap.dedent("""
+            project(consumer)
+            cmake_minimum_required(VERSION 3.1)
+
+            include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+            conan_basic_setup()
+
+            find_package(middle)
+
+            add_executable(main main.cpp)
+            target_link_libraries(main middle::middle)
+            """)
+        main_cpp = textwrap.dedent("""
+            #include "middle.h"
+
+            int main() {
+                middle();
+            }
+            """)
+        client.save({"conanfile.py": conanfile,
+                     "src/CMakeLists.txt": cmakelists,
+                     "src/main.cpp": main_cpp}, clean_first=True)
+        client.run("create . consumer/1.0@")
```

Comment:
Not sure if this last `consumer` adds anything to the test.

Reply:
It is reproducing the case exposed in the issue more clearly, maybe a check to verify that the target link libraries are not set with multiple `nonstd::nonstd` targets is the part missing. But I agree that we can reduce the tests if we don't want to test this one

Reply:
Ok, this is useful to check we are not adding the same target several times.